### PR TITLE
fix: omit ln fee for direct swaps

### PR DESCRIPTION
**Pull request by joschisan**

### CODE REVIEW

File: modules/fedimint-lnv2-client/src/lib.rs [hunk 1]
```diff
@@ -73,10 +73,13 @@ pub struct ReceiveOperationMeta {
     pub contract: IncomingContract,
 }
 
-/// Number of blocks until outgoing lightning contracts time out and user
+/// Number of blocks until outgoing lightning contracts times out and user
 /// client can refund it unilaterally
 pub const EXPIRATION_DELTA_LIMIT_DEFAULT: u64 = 500;
 
+/// A two hour buffer in case either the client or gateway go offline
+pub const CONTRACT_CONFIRMATION_BUFFER: u64 = 12;
```
Comment (elsirion) [inline modules/fedimint-lnv2-client/src/lib.rs:81]: "The client" meaning the user? Is it a problem if they are offline? In LNv1 we explicitly support the user staying offline for extended periods of time while still being able to receive payments on previously generated addresses.

File: gateway/ln-gateway/src/lib.rs [hunk 2]
```diff
@@ -1434,20 +1434,29 @@ impl Gateway {
...
-            // The base fee ensures that the gateway does not loose sats receiving the payment due
-            // to fees paid on the transaction funding the incoming contract
-            receive_fee: PaymentFee::RECEIVE_FEE_LIMIT_DEFAULT,
-            expiration_delta_default: 500,
-            expiration_delta_minimum: EXPIRATION_DELTA_MINIMUM_V2,
-        })
+    pub async fn routing_info_v2(
+        &self,
+        federation_id: &FederationId,
+    ) -> anyhow::Result<Option<RoutingInfo>> {
```
Comment (elsirion) [inline gateway/ln-gateway/src/lib.rs:1440]: Could this just be a `Result<RoutingInfo>`? Is the `None` return value ever being acted upon other than raising an error that the federation doesn't exist?

**Review by elsirion — APPROVED** [review]

File: gateway/ln-gateway/src/lib.rs (hunk 2 above)
Comment (joschisan) [reply gateway/ln-gateway/src/lib.rs:1440]: No but that's the point - federation unsupported is a valid response not an error.

File: modules/fedimint-lnv2-client/src/lib.rs (hunk 1 above)
Comment (joschisan) [reply modules/fedimint-lnv2-client/src/lib.rs:81]: This buffer is only relevant for sending payments, to account for a possible delay between creating an outgoing contract and processing of the contract by the gateway. The expiration delta set by the client will still be accepted for a minimum of 2 hours after creation of the outgoing contract.

Comment (m1sterc001guy) [reply modules/fedimint-lnv2-client/src/lib.rs:81]: Can we name this `CONTRACT_CONFIRMATION_BUFFER_HOURS`? IMO helps to have the unit denomination directly in the name

**Pull request merged**